pymaker/__init__.py

```diff
@@ -63,15 +64,28 @@ def web3_via_http(endpoint_uri: str, timeout=60, http_pool_size=20):
     return Web3(HTTPProvider(endpoint_uri=endpoint_uri, request_kwargs={"timeout": timeout}, session=session))
 
 
-def _is_parity(web3: Web3) -> bool:
+class NonceCalculation(Enum):
+    TX_COUNT = auto()
+    PARITY_NEXTNONCE = auto()
+    SERIAL = auto()
+
+
+def _get_nonce_calc(web3: Web3) -> NonceCalculation:
     assert isinstance(web3, Web3)
-    global node_is_parity
-    if web3 not in node_is_parity:
-        logger.debug(f"node clientVersion={web3.clientVersion}")
-        is_infura = "infura" in web3.manager.provider.endpoint_uri
+    global nonce_calc
+    if web3 not in nonce_calc:
+        providers_without_nonce_calculation = ['infura', 'quiknode']
```

Comment:
There are probably more providers which belong in this list, but I don't have API keys to test.  Perhaps Integrations Team could help with this?

Reply:
We could help with this! I will add it to our backlog. Chainsafe and Alchemy are a few that come to mind; let me know if there're others worth including.

Reply:
I tested on a _Pro_ subscription tier.  If a dedicated key were made available, I could investigate.